akka-stream/src/main/scala/akka/stream/impl/UnfoldResourceSource.scala

```diff
@@ -62,6 +64,7 @@ import scala.util.control.NonFatal
 
     private def restartState(): Unit = {
       close(blockingStream)
+      open = false
```

Comment:
I'm not sure I understand why this couldn't be before the `close`. What happens if `close` throws here? Wouldn't `postStop` be called and then open is still true and another `close` will be done?